File: Daily Challanges/2026-04-April/2026-04-01.js
function fixPrankNumber(nums) {
  const n = nums.length;
  const diffs = [];
  for (let i = 0; i < n - 1; i++) diffs.push(nums[i + 1] - nums[i]);

  const freq = {};
  diffs.forEach((d) => (freq[d] = (freq[d] || 0) + 1));
  const step = Number(Object.entries(freq).sort((a, b) => b[1] - a[1])[0][0]);

  for (let i = 0; i < n - 1; i++) {
    if (diffs[i] !== step) {
      for (const bad of [i, i + 1]) {
        const result = [...nums];
        result[bad] = bad > 0 ? result[bad - 1] + step : result[bad + 1] - step;
        if (
          result.every((_, j) => j === 0 || result[j] - result[j - 1] === step)
        ) {
          return result;
        }
      }
    }
  }

  return [...nums];
}

File: Daily Challanges/2026-04-April/2026-04-01.py
def fix_prank_number(nums):
    n = len(nums)
    diffs = [nums[i + 1] - nums[i] for i in range(n - 1)]
    step = max(set(diffs), key=diffs.count)

    for i in range(n - 1):
        if diffs[i] != step:
            for bad in [i, i + 1]:
                result = nums[:]
                result[bad] = (
                    result[bad - 1] + step if bad > 0 else result[bad + 1] - step
                )
                if all(result[j + 1] - result[j] == step for j in range(n - 1)):
                    return result

    return nums[:]
